meson.build

```diff
@@ -0,0 +1,186 @@
+# Thanks to SciPy project for initial meson.build file!
+project(
+  'pyslalib',
+  'c',
+  version: '1.0',
+  license: 'GPLv2',
+  meson_version: '>= 0.62',
+  default_options: [
+    'buildtype=debugoptimized',
+    'c_std=c99',
+    'c_args=-Wno-unused-function -Wno-conversion -Wno-misleading-indentation -Wno-incompatible-pointer-types -Wunused-variable',
+    'fortran_args=-Wno-conversion -Wunused-dummy-argument -Wno-unused-function -Wunused-variable',
+    'fortran_std=legacy',
+  ],
+)
+
+# Find C/C+ compilers separately from Fortran compiler
+cc = meson.get_compiler('c')
+
+# Adding at project level causes many spurious -lgfortran flags.
+add_languages('fortran', native: false)
+fc = meson.get_compiler('fortran')
+
+# Compiler-dependent arguments
+GCC = cc.get_id() == 'gcc'
+GFORTRAN = fc.get_id() == 'gcc'
+CC_FLAGS = []
+CC_LINK_FLAGS = []
+FC_FLAGS = []
+FC_LINK_FLAGS = []
+
+# TODO: Make flags for VC and Intel compilers
+
+if GCC
+  CC_FLAGS += [
+    '-DDEBUG',
+    '-fopenmp'
+  ]
+  CC_LINK_FLAGS += [
+    '-lgomp',
+    '-lquadmath',
+  ]
+endif
+
+# Although, gfortran has the same flags as gcc, here we specify them
+# separately, because gfortran might be used along with MSVC compiler 
+if GFORTRAN
+  if not GCC
+    CC_FLAGS += [
+      '-DDEBUG',
+      '-fopenmp'
+    ]
+  endif
+  FC_LINK_FLAGS += [
+    '-lgomp',
+    '-lquadmath',
+  ]
+endif
+
+# Platform-dependent flags
+WINDOWS = build_machine.system() == 'windows'
+MINGW = WINDOWS and GCC
+
+if WINDOWS and not GCC
+  error('Currently, libsla could be built only with GCC toolchain on Windows')
+endif
+
+if MINGW
+  # Probably, -DSIZEOF_VOID_P=8 is not neccessarty with -DMS_WIN64
+  MINGW_FLAGS = [
+    '-DSIZEOF_VOID_P=8',
+    '-DMS_WIN64'
+  ]
+  MINGW_LINK_FLAGS = [ 
+    '-static-libgcc', 
+    '-static-libstdc++', 
+    '-static', 
+    '-static-libgfortran',
+  ]
+  CC_FLAGS += MINGW_FLAGS
+  FC_FLAGS += MINGW_FLAGS
+  CC_LINK_FLAGS += MINGW_LINK_FLAGS
+  FC_LINK_FLAGS += MINGW_LINK_FLAGS
+endif
+
+# This argument is called -Wno-unused-but-set-variable by GCC, however Clang
+# doesn't recognize that.
+if cc.has_argument('-Wno-unused-but-set-variable')
+  CC_FLAGS += ['-Wno-unused-but-set-variable']
+endif
+
+# Find python module (src: https://mesonbuild.com/Python-module.html)
+py3 = import('python').find_installation()
+py3_dep = py3.dependency() # In newer versions, Python 3 dependency by default
+py3_venv_dir = run_command(
+  py3.full_path(),
+  ['-c', 
+    'import sys; print(sys.prefix)'
+  ], 
+  check: true).stdout().strip()
+py3_base_dir = run_command(
+  py3.full_path(),
+  ['-c', 
+    'import sys; print(sys.base_prefix)'
+  ], 
+  check: true).stdout().strip()
+
+meson_src_base_dir = meson.current_source_dir()
```

Comment:
Do you really need a constant for this? Instead of `meson.current_source_dir()` and saving it you can use `meson.project_source_root()` anywhere.

Reply:
I use `meson.current_source_dir()` since meson complains about paths to include directories relative to the current source tree. As I realized, this happens, for instance, when you install the package with --no-build-isolation argument and your virtual environment folder located in the repository folder. I haven't found information on if `meson.project_source_root()` does the same in this case.